ValheimPlus/GameClasses/EnvMan.cs

```diff
@@ -25,41 +25,91 @@ public static void SetupDayLength()
             }
         }
 
+        private const float startOfADay = 0.25f;
+        private const float startOfNight = 0.75f;
+        private const float startOfAfternoon = 0.5f;
+
+        /// <summary>
+        /// Return progression of the day on a range of 0 to 1
+        /// </summary>
+        public static double GetDayProgression()
+        {
+            double timeSecondsFromStart = ZNet.instance.GetTimeSeconds();
+            double timeSecondsFromDay = timeSecondsFromStart % Configuration.Current.Time.totalDayTimeInSeconds;
+            return timeSecondsFromDay / Configuration.Current.Time.totalDayTimeInSeconds;
+        }
+
         /// <summary>
         /// Hook on EnvMan init to alter total day length
         /// </summary>
         [HarmonyPatch(typeof(EnvMan), "Awake")]
-        public static class TimeInitHook
+        public static class EnvMan_Awake_Patch
         {
             private static void Prefix(ref EnvMan __instance)
             {
                 SetupDayLength();
             }
         }
 
+        [HarmonyPatch(typeof(EnvMan), "IsDay")]
+        class EnvMan_IsDay_Patch
+        {
+            static bool Postfix(bool result)
+            {
+                if (ZNet.instance.IsServer())
+                {
+                    double progression = GetDayProgression();
+                    result = progression >= startOfADay && progression <= startOfNight;
+                }
+                return result;
+            }
+        }
+
+        [HarmonyPatch(typeof(EnvMan), "IsNight")]
+        class EnvMan_IsNight_Patch
+        {
+            static bool Postfix(bool result)
+            {
+                if (ZNet.instance.IsServer())
+                {
+                    double progression = GetDayProgression();
+                    result = progression <= startOfADay || progression >= startOfNight;
+                }
+                return result;
+            }
+        }
+
+        [HarmonyPatch(typeof(EnvMan), "IsAfternoon")]
+        class EnvMan_IsAfternoon_Patch
+        {
+            static bool Postfix(bool result)
+            {
+                if (ZNet.instance.IsServer())
+                {
+                    double progression = GetDayProgression();
+                    result = progression >= startOfAfternoon && progression <= startOfNight;
+                }
+                return result;
+            }
+        }
+
         /// <summary>
         /// Hook on EnvMan to alter night speed
         /// </summary>
         [HarmonyPatch(typeof(EnvMan), "FixedUpdate")]
-        public static class TimeUpdateHook
+        public static class EnvMan_FixedUpdate_Patch
         {
             private static void Prefix(ref EnvMan __instance)
             {
-                if (Configuration.Current.Time.IsEnabled)
+                if (!Configuration.Current.Time.IsEnabled)
```

Comment:
Do we need to add this check in front of the other patches above as well?

Reply:
Let me add that